### PR TITLE
log additional information into the console

### PR DESCRIPTION
**Pull request by staabm**
print some debug information into the console when the runtime doesnt provide a error.code

### CODE REVIEW

**Comment by staabm** [comment]
I ran into a case where I got a "Unknown error. ". from there it is not easy to debug whats going on. In my case error contained `[Error: Unknown XML-RPC tag 'TITLE']` which might help to find the root-cause of the error.

**Comment by scottgonzalez** [comment]
Did the original error actually help you? If so, how?

**Comment by staabm** [comment]
the original error helped me, because it pointed me into another direction as the message atm does.

my database server was running and wordpress worked propery, but the webserver denied access to the xmlrpc.php because it is used a lot in the www to attack WP instances.

**Comment by staabm** [comment]
alternatively we could utilize the verbose flag, but I guess in this kind of general error it is always helpfull to have all the information available.

**Comment by scottgonzalez** [comment]
How about using [verror](https://www.npmjs.com/package/verror) to expose the original error?

**Comment by staabm** [comment]
I dont have much experience in the NPM ecosystem. In case you like to have a additional dependency for this special case I could add it as well.

**Comment by scottgonzalez** [comment]
verror is fairly small and seems to be pretty popular. I think adding it would be useful.

**Comment by scottgonzalez** [comment]
I thought about this some more. What if instead of creating a new error, we just prepend the current message to the existing error's message and then pass along the original message?

**Comment by scottgonzalez** [comment]
Reading that message again, we'd probably want to append it and drop the "Unknown error" portion. https://github.com/scottgonzalez/node-wordpress/commit/da68666551674d837b45cad6d412a1a69212dcdc should also help in providing more information about what broke.

**Comment by staabm** [comment]
@scottgonzalez something like that?

File: index.js [hunk 1]
```diff
@@ -157,9 +157,15 @@ Client.prototype.validateXmlrpcVersion = function( callback ) {
 					"XML-RPC extensions for Gilded WordPress are not installed." ) );
 			}
 			if ( !error.code ) {
-				return callback( new Error( "Unknown error. " +
+				var msg = "Unknown error. " +
 					"Please ensure that your database server is running " +
-					"and WordPress is functioning properly." ) );
+					"and WordPress is functioning properly."
+
+				if ( error ) {
```
Comment (scottgonzalez) [inline index.js]: We already know that an error exists.

File: index.js [hunk 2]
```diff
@@ -157,9 +157,15 @@ Client.prototype.validateXmlrpcVersion = function( callback ) {
 					"XML-RPC extensions for Gilded WordPress are not installed." ) );
 			}
 			if ( !error.code ) {
-				return callback( new Error( "Unknown error. " +
+				var msg = "Unknown error. " +
```
Comment (scottgonzalez) [inline index.js]: *We* don't know what the error is, but if we're providing the original error object, then there's enough context that we should no longer include this sentence.

File: index.js [hunk 3]
```diff
@@ -159,7 +159,8 @@ Client.prototype.validateXmlrpcVersion = function( callback ) {
 			if ( !error.code ) {
 				return callback( new Error( "Unknown error. " +
```
Comment (scottgonzalez) [inline index.js]: Don't create a new error, that's hiding the information. Just append the text, without "Unknown error" to the existing `message` property.

Comment (staabm) [inline index.js]: like that?

File: index.js [hunk 4]
```diff
@@ -157,9 +157,8 @@ Client.prototype.validateXmlrpcVersion = function( callback ) {
 					"XML-RPC extensions for Gilded WordPress are not installed." ) );
 			}
 			if ( !error.code ) {
-				return callback( new Error( "Unknown error. " +
-					"Please ensure that your database server is running " +
-					"and WordPress is functioning properly." ) );
+				error.message += "Please ensure that your database server is running " +
```
Comment (scottgonzalez) [inline index.js]: This probably needs some whitespace between the existing message and the custom message. Maybe just start this string with a newline?

Comment (staabm) [inline index.js]: thought about that, too. initially I tried to add a space, but the newline is a better approch. fixed.

**Pull request closed without merging**